### PR TITLE
update minimal CMake requirements to 3.11.4

### PR DESCRIPTION
**Pull request by psychocoderHPC**
- [x] update cmake version in examples and `README.md` (CMake requirement is pulled from alpaka)
- [x] update `INSTALL.md`
- [x] mention needed alpaka version

### CODE REVIEW

File: cuplaConfig.cmake [hunk 1]
```diff
@@ -22,7 +22,7 @@
 # Required cmake version.
 ################################################################################
 
-cmake_minimum_required(VERSION 3.11.0)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.11.4)
```
Comment (ax3l) [inline cuplaConfig.cmake]: lowercase is sexy, modern, and consistent with the other CMake code below:
```suggestion
cmake_minimum_required(VERSION 3.11.4)
```

File: example/CUDASamples/asyncAPI/CMakeLists.txt [hunk 2]
```diff
@@ -22,7 +22,7 @@
 # Required CMake version.
 ################################################################################
 
-cmake_minimum_required(VERSION 3.11.0)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.11.4)
```
Comment (ax3l) [inline example/CUDASamples/asyncAPI/CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 3.11.4)
```

File: example/CUDASamples/asyncAPI_tuned/CMakeLists.txt [hunk 3]
```diff
@@ -23,7 +23,7 @@
 # Required CMake version.
 ################################################################################
 
-cmake_minimum_required(VERSION 3.11.0)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.11.4)
```
Comment (ax3l) [inline example/CUDASamples/asyncAPI_tuned/CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 3.11.4)
```

File: example/CUDASamples/matrixMul/CMakeLists.txt [hunk 4]
```diff
@@ -23,7 +23,7 @@
 # Required CMake version.
 ################################################################################
 
-cmake_minimum_required(VERSION 3.11.0)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.11.4)
```
Comment (ax3l) [inline example/CUDASamples/matrixMul/CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 3.11.4)
```

File: example/CUDASamples/vectorAdd/CMakeLists.txt [hunk 5]
```diff
@@ -22,7 +22,7 @@
 # Required CMake version.
 ################################################################################
 
-cmake_minimum_required(VERSION 3.11.0)
+CMAKE_MINIMUM_REQUIRED(VERSION 3.11.4)
```
Comment (ax3l) [inline example/CUDASamples/vectorAdd/CMakeLists.txt]:
```suggestion
cmake_minimum_required(VERSION 3.11.4)
```

**Comment by ax3l** [comment]
@psychocoderHPC please do not forget to also update your `INSTALL.md` with this PR

File: cuplaConfig.cmake (hunk 1 above)
Comment (psychocoderHPC) [reply cuplaConfig.cmake]: copy past copy past copy past^^

**Comment by sbastrakov** [comment]
Note: this will probably close #133.

**Comment by sbastrakov** [comment]
Also: in view of conversation in #129, shouldn't we change travis to 3.11.4 as well, from currently (and probably mistakenly made) 3.15.4?

**Comment by psychocoderHPC** [comment]
> Also: in view of conversation in #129, shouldn't we change travis to 3.11.4 as well, from currently (and probably mistakenly made) 3.15.4?

Oh yes I thought I did it already. Will ship a PR tomorrow.

**Pull request merged**